File: LinkShortener/LinkShortener.API/Models/PaginatedData.cs
using System.Collections.Generic;

namespace LinkShortener.API.Models
{
    public class PaginatedData<T>
    {
        public IEnumerable<T> Data { get; set; }
        public int TotalCount { get; set; }
        public int Page { get; set; }
        public int Size { get; set; }
    }
}

File: LinkShortener.Site/src/app/_models/paginated-data.ts
export class PaginatedData<T> {
    data: T[];
    totalCount: number;
    page: number;
    size: number;

    constructor(json: any) {
        this.data = json.data;
        this.page = json.page;
        this.size = json.size;
        this.totalCount = json.totalCount;
    }
}